common/packet/include/packet.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef PACKET_H
+#define PACKET_H
+#include <stdio.h>
+#include <iostream>
+#include "glm/glm.hpp"
+#include <string.h>
+#include <deque>
+
+class Packet {
+
+   public:
+    std::deque<u_int8_t> buffer;
+
+    void write_byte(char data);
+    void write_int(int data);
+    void write_float(float data);
+    void write_double(double data);
+    void write_vec3(glm::vec3 data);
+
+    // return -1 if error, else return size
+    int read_byte(char* dest);
+    int read_int(int* dest);
+    int read_float(float* dest);
+    int read_double(double* dest);
+    int read_vec3(glm::vec3* dest);
+
+    u_int8_t* getBytes();
```

Comment:
Make sure you change every instance of u_int8_t to uint8_t, u_int32_t to uint32_t, etc.